kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -151,117 +156,109 @@ function redirectToExploreChannel(store) {
-          store.dispatch('CORE_SET_PAGE_LOADING', false);
-          store.dispatch('CORE_SET_ERROR', null);
-        },
-        (error) => {
-          store.dispatch('CORE_SET_ERROR', JSON.stringify(error, null, '\t'));
-          store.dispatch('CORE_SET_PAGE_LOADING', false);
-        }
-      );
     },
     (error) => {
       store.dispatch('CORE_SET_ERROR', JSON.stringify(error, null, '\t'));
       store.dispatch('CORE_SET_PAGE_LOADING', false);
-    });
+      document.title = _errorTitle();
+    }
+  );
 }
 
 
-function showExploreTopic(store, channelId, id) {
+function showExploreTopic(store, channelId, id, isRoot = false) {
   store.dispatch('CORE_SET_PAGE_LOADING', true);
-  store.dispatch('SET_PAGE_NAME', PageNames.EXPLORE_TOPIC);
-  store.dispatch('SET_CURRENT_CHANNEL', channelId);
-  cookiejs.set('currentChannel', channelId);
+  if (isRoot) {
+    store.dispatch('SET_PAGE_NAME', PageNames.EXPLORE_CHANNEL);
+  } else {
+    store.dispatch('SET_PAGE_NAME', PageNames.EXPLORE_TOPIC);
+  }
 
-  const attributesPromise = ContentNodeResource.getModel(id).fetch();
+  const topicPromise = ContentNodeResource.getModel(id).fetch();
   const childrenPromise = ContentNodeResource.getCollection({ parent: id }).fetch();
-  _updateChannelList(store);
-  ConditionalPromise.all([attributesPromise, childrenPromise]).only(
+  const channelsPromise = ChannelResource.getCollection({}).fetch();
+  ConditionalPromise.all([topicPromise, childrenPromise, channelsPromise]).only(
     samePageCheckGenerator(store),
-    ([attributes, children]) => {
-      const pageState = { id };
-      pageState.topic = _topicState(attributes);
+    ([topic, children, channelsData]) => {
+      if (!_handleChannels(store, channelId, _channelListState(channelsData))) {
+        return;
+      }
+      const pageState = {};
+      pageState.topic = _topicState(topic);
       const collection = _collectionState(children);
       pageState.subtopics = collection.topics;
       pageState.contents = collection.contents;
       store.dispatch('SET_PAGE_STATE', pageState);
       store.dispatch('CORE_SET_PAGE_LOADING', false);
       store.dispatch('CORE_SET_ERROR', null);
+      if (isRoot) {
+        const currentChannel = getters.currentChannel(store.state);
+        document.title = _explorePageTitle(currentChannel.title);
+      } else {
+        document.title = _explorePageTitle(pageState.topic.title);
+      }
     },
     (error) => {
       store.dispatch('CORE_SET_ERROR', JSON.stringify(error, null, '\t'));
       store.dispatch('CORE_SET_PAGE_LOADING', false);
+      document.title = _errorTitle();
+    }
+  );
+}
+
+
+function showExploreChannel(store, channelId) {
+  store.dispatch('CORE_SET_PAGE_LOADING', true);
+  store.dispatch('SET_PAGE_NAME', PageNames.EXPLORE_CHANNEL);
+
+  ChannelResource.getCollection({}).fetch().then(
+    (channelsData) => {
+      if (!_handleChannels(store, channelId, _channelListState(channelsData))) {
```

Comment:
Should definitely change this to a check, I think? Could then do (or this could be abstracted further):
```
if (!_checkChannels(store, channelId, _channelListState(channelsData))) {
    handleError('Channel not Found');
    return;
}
```